tracer/src/Datadog.Trace/DataStreamsMonitoring/Utils/BinaryPrimitivesHelper.cs

```diff
@@ -84,4 +86,82 @@ public static ulong ReadUInt64LittleEndian(byte[] bytes)
         }
 #endif
     }
+
+#if NETCOREAPP3_1_OR_GREATER
+
+    /// <summary>
+    /// Reverses a primitive value - performs an endianness swap
+    /// </summary>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static ulong ReverseEndianness(ulong value)
+    {
+        return System.Buffers.Binary.BinaryPrimitives.ReverseEndianness(value);
+    }
+
+#else
+
+    /// <summary>
+    /// Reverses a primitive value - performs an endianness swap
+    /// </summary>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static ulong ReverseEndianness(ulong value)
+    {
+        // Operations on 32-bit values have higher throughput than
+        // operations on 64-bit values, so decompose.
+        return ((ulong)ReverseEndianness((uint)value) << 32) + ReverseEndianness((uint)(value >> 32));
+    }
+
+    /// <summary>
+    /// Reverses a primitive value - performs an endianness swap
+    /// </summary>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static uint ReverseEndianness(uint value)
+    {
+        // This takes advantage of the fact that the JIT can detect
+        // ROL32 / ROR32 patterns and output the correct intrinsic.
```

Comment:
_Maybe_ worth adding a comment that this logic and reasoning comes from CoreCLR

Reply:
Done, thanks.